libglusterfs/src/glusterfs/common-utils.h

```diff
@@ -1245,18 +1245,18 @@ gf_tvdiff(struct timeval *start, struct timeval *end)
 
 /* Return delta value in nanoseconds. */
 
-static inline double
+static inline uint64_t
 gf_tsdiff(struct timespec *start, struct timespec *end)
 {
     struct timespec t;
 
     if (start->tv_nsec > end->tv_nsec)
-        t.tv_sec = end->tv_sec - 1, t.tv_nsec = end->tv_nsec + 1000000000;
+        t.tv_sec = end->tv_sec - 1, t.tv_nsec = end->tv_nsec + GF_SEC_IN_NS;
     else
         t.tv_sec = end->tv_sec, t.tv_nsec = end->tv_nsec;
 
-    return (double)(t.tv_sec - start->tv_sec) * 1e9 +
-           (double)(t.tv_nsec - start->tv_nsec);
+    return (uint64_t)(t.tv_sec - start->tv_sec) * GF_SEC_IN_NS +
```

Comment:
nit: this looks OK. In the next iteration (if we'll have it), I assume we'd want to get rid of 't' altogether, as it's not really needed here. You could return one value in line 1254 and a different path in line 1256, without the assignments.

But this should not hold this patch.